crates/toml/src/de.rs

```diff
@@ -45,6 +45,138 @@ where
     T::deserialize(Deserializer::new(s))
 }
 
+/// Deserializes a reader into a type.
+///
+/// This function will attempt to interpret the reader as a TOML document and
+/// deserialize `T` from the document.
+///
+/// To deserializes TOML values, instead of documents, see [`ValueDeserializer`].
+///
+/// # Examples
+///
+/// ```
+/// use serde::Deserialize;
+/// #[derive(Deserialize)]
+/// struct Config {
+///    title: String,
+///   owner: Owner,
+/// }
+///
+/// #[derive(Deserialize)]
+/// struct Owner {
+///    name: String,
+/// }
+///
+/// let config: Config = toml::from_reader(r#"
+///     title = 'TOML Example'
+///     [owner]
+///     name = 'Lisa'
+/// "#.as_bytes()).unwrap();
+///
+/// assert_eq!(config.title, "TOML Example");
+/// assert_eq!(config.owner.name, "Lisa");
+/// ```
+#[cfg(feature = "parse")]
+pub fn from_reader<R, T>(rdr: R) -> Result<T, Error>
```

Comment:
We should probably have tests for these functions beyond the doctests

For example., `from_value` is using the table deserializer, rather than the value deserializer

Reply:
`from_value` has been put on hold.

Are there any items of concern in other functions?
I thought the test cases for the existing methods in toml were sufficient.